packages/domain-search/src/components/search-bar/filter.tsx

```diff
@@ -0,0 +1,124 @@
+import { Dropdown } from '@wordpress/components';
+import { useCallback, useMemo, useState } from 'react';
+import { useDomainSearch } from '../../page/context';
+import { FilterState } from '../../page/types';
+import { DomainSearchControls } from '../../ui';
+import type { TokenItem } from '@wordpress/components/build-types/form-token-field/types';
+
+const emptyFilter: FilterState = {
+	exactSldMatchesOnly: false,
+	tlds: [],
+};
+
+type Props = {
+	onSubmit?: () => void;
+	showTldFilter: boolean;
```

Comment:
I don't understand the purpose of this `showTldFilter` variable. Do we need that? I feel that if the tlds array is empty within the UI component, then surely we don't want to render anything? It seems we're erroring on the side of redundancy instead of just taking decisions based on derived data.